src/plugins/intel_cpu/src/nodes/generate_proposals.cpp

```
@@ -477,6 +380,23 @@ bool GenerateProposals::needPrepareParams() const {
     return false;
 }
 
+std::unique_ptr<jit_uni_nms_proposal_kernel> GenerateProposals::compileNMSKernel() {
+    jit_uni_nms_proposal_kernel::jit_nms_conf jcp { post_nms_topn_, nms_thresh_,
+        coordinates_offset_ };
+    std::unique_ptr<jit_uni_nms_proposal_kernel> nms_kernel;
+    if (mayiuse(avx512_core)) {
+        nms_kernel.reset(new jit_uni_nms_proposal_kernel_impl<avx512_core> { jcp });
+    } else if (mayiuse(x64::avx2)) {
+        nms_kernel.reset(new jit_uni_nms_proposal_kernel_impl<x64::avx2> { jcp });
+    } else if (mayiuse(sse41)) {
+        nms_kernel.reset(new jit_uni_nms_proposal_kernel_impl<sse41> { jcp });
+    } else {
+        IE_THROW() << "Unsupported ISA";
```

Why throw exception instead of returning empty `std::unique_ptr<jit_uni_nms_proposal_kernel>` ?

---

`compileNMSKernel()` method was removed in sake of `createPrimitive()` implementation.